dts/bindings/stepper/zephyr,step-direction-stepper-common.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2024 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    This binding provides a stepper controller that can be used with a stepper motor that
+    accepts step and direction signals.
+
+include: base.yaml
+
+properties:
+  step-gpios:
+    type: phandle-array
+    description: |
+      The GPIO pins that are used to send step signals to the stepper motor.
+
+  direction-gpios:
+    type: phandle-array
+    description: |
+      The GPIO pins that are used to send direction signals to the stepper motor.
+      Pin will be driven high for forward direction and low for reverse direction.
+
+  dual-edge-step:
+    type: boolean
+    description: |
+      If present, the stepper motor controller supports dual edge step signals.
```

Comment:
So step on falling and rising edge? actually, we should probably have an enum for this:
```
step-edges:
  type: string
  description: "Edges on step-gpios which motivate stepping"
  enum:
    - FALLING
    - RISING
    - BOTH
```

Reply:
Hmm will there be any real difference between FALLING and RISING though? For both of those you have to perform two toggles, or am I missing something?

Reply:
The description is basically just expanding the binding into a sentence.

Just read in the datasheet, that basically this feature cuts toggle rate in half. This feature is supported by tmc drivers, however, i have not seen this kind of support in drivers such as a4988 and drv882x series. Probably for now it would be more apt to put this tmc22xx bindings? Also this setting would have to take place via UART, or? I mean low end step-dir drivers dont normally support such sampling on both the edges  afaik.

Reply:
> Just read in the datasheet, that basically this feature cuts toggle rate in half. This feature is supported by tmc drivers, however, i have not seen this kind of support in drivers such as a4988 and drv882x series. Probably for now it would be more apt to put this tmc22xx bindings? Also this setting would have to take place via UART, or? I mean low end step-dir drivers dont normally support such sampling on both the edges afaik.

Yeah but since the stepping is done by the "common" code this sort of information must also live in the common structure.

> And the other two bindings namely step-gpios & dir-gpios could be placed in stepper-controller.yaml since a lot of drivers, for instance, even tmc5xxx series also has that interface.

Yeah good call will move the properties to the stepper-controller.yaml instead :^)

Reply:
Removed the `zephyr,step-direction-stepper-common.yaml` and moved them into the `stepper-controller.yaml` instead and the added the `dual-edge-step` to the tmc binding instead.